jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy

```diff
@@ -272,12 +272,22 @@ def step075_customize_puppet_settings(script_dir, puppet_settings) {
     }
 }
 
-def step080_customize_java_args(script_dir, server_java_args, server_era) {
-    if ((server_java_args == null) || (server_java_args == "")) {
+def step080_customize_java_args(script_dir, server_heap_settings, server_era) {
+    if ((server_heap_settings == null || server_heap_settings == "") && "${JAVA_ARGS_ADDITIONS}" == "" && "${HEAP_OVERRIDES}" == "") {
         echo "Skipping java_args configuration because none specified in job"
     } else {
+        if (server_heap_settings == null) {
+            server_heap_settings = ""
+        }
+
+        if ("${HEAP_OVERRIDES}" != "" && "${HEAP_OVERRIDES}" != null) {
```

Comment:
We check that the `HEAP_OVERRIDES` here are not null but above we only check to see if they are an empty string. Can we always make the assumption that we make in the first condition (params will always be a string) and remove this null check, or should we also add null checks above?

Reply:
Mmmm... Does the interpolation (`"${HEAP_OVERRIDES}"`) mean that if the variable is null it will be a blank string?

Reply:
Oh the null check was me trying to figure out what was wrong. But... null interpolated will not be the empty string. it will be "null".

Reply:
so i can pull out the heap_overrides null check because it won't do the right thing at all.